src/test/java/alfio/controller/api/admin/EventApiControllerIntegrationTest.java

```diff
@@ -202,13 +215,22 @@ private Pair<Event,String> createEvent(Event.EventFormat format) {
 
     }
 
+    private void createPromoCode(int organizationId, String email) {
+        var eventId = event.getId();
+        promoCodeDiscountRepository.addPromoCode(TEST_PROMO_CODE, eventId, organizationId, ZonedDateTime.now(), ZonedDateTime.now()
+                                                                                                                             .plusDays(3), 10, PromoCodeDiscount.DiscountType.FIXED_AMOUNT, "[1,2,3]", 1, "test promo code", "test-email@gmail.com", PromoCodeDiscount.CodeType.DISCOUNT, 21, "usd");
+        var promoCodeId = promoCodeDiscountRepository.findAllInEvent(eventId).get(0).getId();
+        ticketReservationRepository.setPromotionCodeDiscountId(email, promoCodeId);
```

Comment:
See above. You can run this update using NamedParameterJdbcTemplate. See for example [DemoModeDataManagerIntegrationTest](https://github.com/alfio-event/alf.io/blob/main/src/test/java/alfio/manager/DemoModeDataManagerIntegrationTest.java#L101)